### PR TITLE
Use appropriate log level for worker console logs for v3

### PR DESCRIPTION
**Pull request by ejizba**
Currently, worker console logs are always categorized as "Debug" even though they can contain useful warnings or errors. For example, the Node.js worker displays warnings/errors when validating the Node.js version prior to initializing the rpc channel.

This may be a regression caused by https://github.com/Azure/azure-functions-host/pull/6576. That PR doesn't mention anything about changing log level, so I'm pretty sure it was unintentional, but cc @pragnagopa @AnatoliB to confirm

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs
Comment (pragnagopa) [inline src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs:193]: This was an intentional change - as we should consider all Worker logs as System Logs - system logs are noisy for Cx but Debug is the default logLevel for Kusto - product team has access to all debug level system logs.

**Review by pragnagopa — CHANGES_REQUESTED** [review]
If you do need few system logs to be logged at Info level - we need to come up a distinction for System Log - Info level. I recommend leaving the default loglevel here to be Debug.

File: src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs
Comment (ejizba) [reply src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs:193]: Aren't those two separate things? I agree they are system logs, but normally I see us choose a level _and_ a category. The level can be "Error", "Warning", "Information", "Debug", etc. and the category can be either "System" or "User". In this case I thought `_workerProcessLogger` was used for system logs and `_consoleLogSource` (in the "else" case below) was used for user logs.

Without proper levels, it makes supportability more difficult for the workers because any warning/error the worker logs is ignored - users _can't_ see it and the support team _won't_ see it because they often filter by log level.

Finally - at least when I'm running locally the default log level for the "Worker" category is "warning", so it's not noisy at all in my experience.

Comment (fabiocav) [reply src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs:193]: Interesting. @pragnagopa it does look like the change has an impact on the logs surfaced to the end user, which is something we're trying to address. Is it possible the original change had a broader impact than anticipated and left us with a gap that makes troubleshooting more difficult for customers?

Comment (pragnagopa) [reply src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs:193]: Looking at the previous PR and the test added, the intention was to force language worker logs to be treated as  system logs similar to what we do in the host.  Also, condition here ` if (WorkerProcessUtilities.IsConsoleLog(msg))` - is specifically to ensure the log has known prefix.

If you would like to go ahead with this change - I recommend scoping the change to language workers that the team owns - as of now they are shipped and packaged with the host - to avoid logging info level logs from language workers not owned by the team.

Comment (liliankasem) [reply src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs:193]: That sounds good to me, I would like to go ahead with the change if possible - @ejizba is it possible to scope this as Pragna suggested?

**Comment by AnatoliB** [comment]
cc @gzuber @RohitRanjanMS - FYI

File: src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs
Comment (pragnagopa) [reply src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs:193]: @ejizba - I do not think host has a list of language workers that are well known - open a separate for the recommendation above. Approving this PR!

**Review by pragnagopa — APPROVED** [review]
Please do wait for final approval from @fabiocav / @liliankasem

**Review by liliankasem — APPROVED** [review]

File: src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs
Comment (ejizba) [reply src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs:193]: Filed this issue to follow up: https://github.com/Azure/azure-functions-host/issues/8109

**Pull request merged**
